037-Top-K-Frequent-Elements.py

```diff
@@ -0,0 +1,24 @@
+'''
+Given a non-empty array of integers, return the k most frequent elements.
+
+Example 1:
+
+Input: nums = [1,1,1,2,2,3], k = 2
+Output: [1,2]
+
+Example 2:
+
+Input: nums = [1], k = 1
+Output: [1]
+
+Note:
+
+    You may assume k is always valid, 1 ≤ k ≤ number of unique elements.
+    Your algorithm's time complexity must be better than O(n log n), where n is the array's size.
+'''
+class Solution:
+    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
+        count = {}
+        for num in nums:
+            count[num] = count.get(num, 0) + 1
+        return heapq.nlargest(k, count.keys(), key = count.get)
```

Comment:
Time and Space Complexity?

Reply:
Time Complexity: O(n log(k))
Space Complexity: O(n)